src/Microsoft.Extensions.Options/OptionsService.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System;
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Implementation of IOptionsFactory.
+    /// </summary>
+    /// <typeparam name="TOptions">The type of options being requested.</typeparam>
+    public class OptionsService<TOptions> : IOptionsService<TOptions> where TOptions : class, new()
```

Comment:
You can turn it off specifically for an options type by closing the generic, i.e. `services.AddSingleton<IOptionsCache<NoCacheOptions>, NoCache<NoCacheOptions>>()`